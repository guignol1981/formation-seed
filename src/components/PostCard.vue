<template>
    <m-panel
        class="post"
        :highlighted="true"
        :padding="false"
        :shadow="true"
        skin="light"
    >
        <div slot="header" class="post__header">
            <h2 class="post__title">{{ post.title }}</h2>
        </div>
        <div class="post__body" v-html="post.body"></div>
        <div slot="footer" class="post__footer">
            <m-link>#DÃ©tails</m-link>
            <m-icon-button
                iconName="m-svg__edit"
                buttonSize="32px"
                iconSize="20px"
                skin="light"
            ></m-icon-button>
        </div>
    </m-panel>
</template>

<script lang="ts">
    import Vue from 'vue';
    import Component from 'vue-class-component';
    import { Prop, Emit } from 'vue-property-decorator';

    @Component
    export default class PostCard extends Vue {
        @Prop({ required: true })
        public post!: { id: string; title: string; body: string };

        @Emit('editClicked')
        public emitEditClicked(postId: string): void {}

        protected created(): void {
            this.emitEditClicked(this.post.id);
        }
    }
</script>

<style lang="scss" scoped>
    @import '~@ulaval/modul-components/dist/styles/commons';

    .post {
        position: relative;

        + .post {
            margin-top: $m-spacing--l;
        }

        &__header {
            display: flex;
            align-items: center;
            font-weight: bold;
            font-size: $m-font-size--h3;
            padding: 1rem;

            &__author {
                font-size: $m-font-size--s;
            }
        }

        &__title {
            margin: 0 0 0 $m-spacing--s;

            @media (min-width: $m-mq--min--s) {
                margin: 0 0 0 $m-spacing--m;
            }
        }

        &__profile-image {
            flex-shrink: 0;
            width: 30px;
            height: 30px;
            border-radius: 50%;
            overflow: hidden;
            background: $m-color--grey;

            @media (min-width: $m-mq--min--s) {
                width: 44px;
                height: 44px;
            }

            img {
                width: 100%;
            }
        }

        &__body {
            text-align: start;
            padding: $m-spacing--m;
            min-height: $m-spacing--xl;
        }

        &__footer {
            display: flex;
            justify-content: space-between;
            align-items: center;
            background-color: $m-color--white;
            padding: $m-spacing--s !important;

            & > div > * {
                margin-left: $m-spacing--s;
            }
        }
    }
</style>
