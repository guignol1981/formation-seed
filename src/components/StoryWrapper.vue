<template>
    <div class="m-u--app-body">
        <slot></slot>
    </div>
</template>

<script lang="ts">
    import Vue from 'vue';
    import Component from 'vue-class-component';

    @Component
    export default class StoryWrapper extends Vue {}
</script>

<style lang="scss">
    $m-font-path: '~@ulaval/modul-components/dist/assets/fonts/'; //this is required.
    @import '~@ulaval/modul-components/dist/styles/main.scss';
</style>
