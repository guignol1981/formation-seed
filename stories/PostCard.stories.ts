import { Component } from 'vue';
import PostCard from '../src/components/PostCard';
import StoryWrapper from '../src/components/StoryWrapper';
import StoryRouter from 'storybook-vue-router';
import { action } from '@storybook/addon-actions';
import { object } from '@storybook/addon-knobs';

export default {
    title: 'PostCard',
    component: PostCard,
    decorators: [StoryRouter()]
};

export const Default: () => Component = (): Component => ({
    components: { PostCard, StoryWrapper },
    data: (): {} => {
        return {
            post: {
                id: '1',
                title: 'An intersting post title',
                body: 'An even more interesting body'
            }
        };
    },
    template: `
    <StoryWrapper>
            <PostCard 
                :post="post" 
                @editClicked="onEditClicked($event)"
            >
            </PostCard>
    </StoryWrapper>
    `,
    methods: {
        onEditClicked: action('editClicked')
    }
});
