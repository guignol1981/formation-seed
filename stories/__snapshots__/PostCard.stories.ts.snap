// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Card Default 1`] = `
<div
  class="m-u--app-body"
>
  <article
    class="m-panel post m--is-skin-light m--has-shadow m--has-border m--is-highlighted"
    data-v-74fdcb1f=""
  >
    <header
      class="m-panel__header m--no-padding"
      data-v-74fdcb1f=""
    >
      <div
        class="m-panel__header__content"
        data-v-74fdcb1f=""
      >
        <div
          class="post__header"
          data-v-74fdcb1f=""
        >
          <h2
            class="post__title"
            data-v-74fdcb1f=""
          >
            An intersting post title
          </h2>
        </div>
      </div>
       
      <!---->
       
      <!---->
    </header>
     
    <div
      class="m-panel__body m--no-padding"
      data-v-74fdcb1f=""
    >
       
      <div
        class="post__body"
        data-v-74fdcb1f=""
      >
        An even more interesting body
      </div>
       
    </div>
     
    <footer
      class="m-panel__footer m--no-padding"
      data-v-74fdcb1f=""
    >
      <div
        class="post__footer"
        data-v-74fdcb1f=""
      >
        <a
          class="m-link router-link-exact-active router-link-active m--has-content"
          data-v-74fdcb1f=""
          data-v-b5057c2e=""
          href="#/"
          tabindex="0"
        >
          <!---->
           
          <span
            class="m-link__text"
            data-v-b5057c2e=""
          >
            #DÃ©tails
          </span>
           
          <!---->
        </a>
         
        <button
          class="m-icon-button m--is-skin-light m--has-ripple"
          data-v-4b160bdb=""
          data-v-74fdcb1f=""
          style="width: 32px; height: 32px;"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="m-icon m-icon-button__icon"
            data-v-3d21e689=""
            data-v-4b160bdb=""
            height="20px"
            width="20px"
          >
            <!---->
             
            <use
              aria-hidden="true"
              class=""
              data-v-3d21e689=""
              xlink:href="#m-svg__edit"
            />
          </svg>
           
          <!---->
        </button>
      </div>
    </footer>
  </article>
</div>
`;
